app/src/util/pdf_utils.py

```diff
@@ -9,6 +8,41 @@
 from pdfminer.psparser import PSLiteral
 
 
+def as_pdf_doc(pdf: BinaryIO | PDFDocument) -> PDFDocument:
+    if isinstance(pdf, PDFDocument):
+        return pdf
+    else:
+        return PDFDocument(PDFParser(pdf))
+
+
+def get_pdf_info(pdf: BinaryIO | PDFDocument, count_pages: bool = False) -> dict[str, Any]:
+    doc = as_pdf_doc(pdf)
+    assert len(doc.info) == 1
```

Comment:
Is there a reason to throw an exception if this fails?

Reply:
The assertion should throw an AssertionError if it's not true. I'll add an error message.